test/errors.js

```diff
@@ -0,0 +1,304 @@
+'use strict';
+
+let { lintLocationMarker: M, positioned, assertError } = require('./utils.js');
+
+describe('errors', () => {
+  it('no contributors', async () => {
+    await assertError(
+      '',
+      {
+        ruleId: 'no-contributors',
+        nodeType: 'html',
+        message:
+          'contributors not specified, skipping copyright boilerplate. specify contributors in your frontmatter metadata',
+      },
+      {
+        copyright: true,
+      }
+    );
+  });
+
+  it('metadata failed', async () => {
+    await assertError(
+      positioned`
+      ${M}<pre class=metadata>
+        this isn't valid yaml: "
+      </pre>
+      `,
+      {
+        ruleId: 'invalid-metadata',
+        nodeType: 'pre',
+        message: 'metadata block failed to parse',
+      }
+    );
+  });
+
+  it('could not find replacement algorithm target', async () => {
+    await assertError(
+      positioned`
+      <emu-alg replaces-step="${M}missing">
+        1. Foo.
+      </emu-alg>
+      `,
+      {
+        ruleId: 'invalid-replacement',
+        nodeType: 'emu-alg',
+        message: 'could not find step "missing"',
+      }
+    );
+  });
+
+  it('replacement algorithm targets something other than a step', async () => {
+    await assertError(
+      positioned`
+      <emu-clause id="clause">
+      <h1>Title</h1>
+        This is a clause.
+      </emu-clause>
+      <emu-alg replaces-step="${M}clause">
+        1. Foo.
+      </emu-alg>
+      `,
+      {
+        ruleId: 'invalid-replacement',
+        nodeType: 'emu-alg',
+        message: 'expected algorithm to replace a step, not a clause',
+      }
+    );
+  });
+
+  it('cycle in replacement algorithms', async () => {
```

Comment:
Can you add a bigger (non-self-referential) cycle test?

Reply:
Done.